mmv1/products/vertexai/IndexEndpointDeployedIndex.yaml

```diff
@@ -0,0 +1,268 @@
+update_verb: :POST
+delete_verb: :POST
+references: !ruby/object:Api::Resource::ReferenceLinks
+  api: https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.indexEndpoints#DeployedIndex
+async: !ruby/object:Api::OpAsync
+  actions:
+    - create
+    - delete
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 45
+      update_minutes: 45
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+    resource_inside_response: true
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+description: |-
+  An endpoint indexes are deployed into. An index endpoint can have multiple deployed indexes.
+autogen_async: false
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "vertex_ai_index_endpoint_deployed_index_basic"
+    primary_resource_id: "basic_deployed_index"
+    vars:
+      project: 'vertex-ai'
+      endpoint_name: 'endpoint-name'
+      network_name: 'network-name'
+      display_name: "vertex-deployed-index"
+      display_name_index: 'test-index'
+      bucket_name: "bucket-name"
+      address_name: 'vertex-ai-range'
+    test_vars_overrides:
+      network_name: 'acctest.BootstrapSharedServiceNetworkingConnection(t, "vpc-network-1")'
+      address_name: 'acctest.BootstrapSharedTestGlobalAddress(t, "vpc-network-1", acctest.AddressWithPrefixLength(8))'
+  - !ruby/object:Provider::Terraform::Examples
+    name: "vertex_ai_index_endpoint_deployed_index_dedicated_resources"
+    primary_resource_id: "dedicated_resources"
+    vars:
+      display_name_index: 'test-index'
+      bucket_name: "bucket-name"
+      display_name: "vertex-deployed-index"
+    skip_docs: true
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  encoder: templates/terraform/encoders/vertex_ai_index_endpoint_deployed_index.go.erb
+  update_encoder: templates/terraform/update_encoder/vertex_ai_index_endpoint_deployed_index.go.erb
+  decoder: templates/terraform/decoders/vertex_ai_index_endpoint_deployed_index.go.erb
+  pre_delete: templates/terraform/pre_delete/vertex_ai_index_endpoint_deployed_index.go.erb
+parameters:
+  - !ruby/object:Api::Type::String
+    name: region
+    description: The region of the index endpoint. eg us-central1
+    url_param_only: true
+    immutable: true
+  - !ruby/object:Api::Type::ResourceRef
+    name: 'indexEndpoint'
+    description: |
+      Identifies the index endpoint. Must be in the format
+      'projects/{{project}}/locations/{{region}}/indexEndpoints/{{indexEndpoint}}'
+    required: true
+    immutable: true
+    resource: 'IndexEndpoint'
+    imports: 'name'
+    url_param_only: true
```

Comment:
What's your preference here? The options are:

- Schema includes project, region and index_endpoint where index_endpoint is a short name
- Schema doesn't include project or region, and only accepts index_endpoint of the format projects/{{project}}/locations/{{region}}/indexEndpoints/{{indexEndpoint}}, thereby providing the region and project values.
    - An example of this elsewhere in the provider is [google_kms_crypto_key](https://github.com/GoogleCloudPlatform/magic-modules/blob/24f92f9cd64e2829ef3e3fd4e7138594f0b5f56b/mmv1/products/kms/CryptoKey.yaml)

Reply:
After thinking through this it makes sense to go with the `projects/{{project}}/locations/{{region}}/indexEndpoints/{{indexEndpoint}}` format. Would make it easier for users since all they would need is to use the id of the `indexEndpoint` resource. Since this `deployed_index` resource depends on an index_endpoint existing already. It would use the `project` and `region` values previously set by `indexEndpoint`. We know that it depends on it since it requires a `indexEndpoint` link in order to `:deployIndex`.